### DISCUSSION TITLE
Is it possible to change token names?

### DISCUSSION THREAD

**Discussion by opoplawski** [p11-glue/p11-kit#634] [Q&A]
We're making use of YKs.  Previously our certificates contained a subject name and p11-kit would use that for the token name (e.g. `Orion%20Poplawski`).  We're looking to move to certificates that do not contain a subject name to avoid credential caching issues on Windows, but with that change the token name seems to become `PIV_II`.  Is there any way to affect this?  It would be nice to have something like `YubiKey`.

**Comment by ueno** [comment]
AFAIK, at the PKCS#11 level where p11-kit operates on, the only way to rename a token is to re-initialize it. I'm not sure if the tools provided by Yubico has the ability nor if `PIV_II` is hard-coded, though we can certainly add a command to initialize a token, say `p11-kit init-token` if that helps.

**Reply by opoplawski** [reply]
Are there some lower level commands I can use to inspect where the `PIV_II` name is coming from?.  Thanks.
